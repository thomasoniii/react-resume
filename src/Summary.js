import React from 'react';

export default ({summary}) => {
  return (
    <div className='container-fluid section'>
      <div className='row'>
        <div className='col section-header'>Summary</div>
      </div>
      <div className='row'>
        <div className='col'>
          {summary}
        </div>
      </div>
    </div>
  )
};
