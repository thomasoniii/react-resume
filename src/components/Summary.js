import React from "react";

const Summary = ({ summary }) => {
  return (
    <div className="container-fluid section">
      <div className="row">
        <div className="col section-header">Summary</div>
      </div>
      <div className="row">
        <div className="col">
          {summary.map((p, i) => (
            <p key={i}>{p}</p>
          ))}
        </div>
      </div>
    </div>
  );
};

export default React.memo(Summary);
