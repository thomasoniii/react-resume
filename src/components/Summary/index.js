import React from "react"
import PropTypes from "prop-types"
import { Card, Typography } from "antd"
const { Title } = Typography

const Summary = ({ summary }) => (
  <Card size="small" title={<Title level={2}>Summary</Title>}>
    {summary.map((p, i) => (
      <p key={i}>{p}</p>
    ))}
  </Card>
)

Summary.propTypes = { summary: PropTypes.object }

export default React.memo(Summary)
