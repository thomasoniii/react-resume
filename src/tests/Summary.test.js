import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"

import Summary from "../components/Summary"
import resume from "../data/resume-example.json"

// <Summary    filters={ filters } summary={resume.summary}       key='summary'/>

const filters = {}

describe("Education tests", () => {
  test("Summary exists properly", () => {
    const summary = renderer.create(
      <Summary filters={filters} summary={resume.summary} />
    )

    const tree = summary.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
