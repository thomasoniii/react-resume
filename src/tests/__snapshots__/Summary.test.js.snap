// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education tests Summary exists properly 1`] = `
<div
  className="container-fluid section"
>
  <div
    className="row"
  >
    <div
      className="col section-header"
    >
      Summary
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        As you can see, I am able to walk on water while playing the ukelele.
      </p>
      <p>
        Your summary section should sum you up. Type in whatever is useful and interesting here to sell yourself.
      </p>
      <p>
        Each value in the array goes into a new paragraph.
      </p>
      <p>
        Tell the employer what they want to know! Don't be afraid to use filters to make a custom resume tailored to a job!
      </p>
    </div>
  </div>
</div>
`;
